### PR TITLE
Ensure token module only supports lsk token

### LINKED ISSUE #8471: Token module supports all tokens from mainchain and not just LSK
### Expected behavior
the supported token should only accept LSK specifically as defined in the [LIP](https://github.com/LiskHQ/lips/blob/3acef71ed72a521de73e4e173932490c9eacbd20/proposals/lip-0051.md)

### Actual behavior
https://github.com/LiskHQ/lisk-sdk/blob/89e7504ef5eb6183aefe576a93be3d6052e56038/framework/src/modules/token/stores/supported_tokens.ts#L192-L199

is accepting all token on the mainchain, not only LSK which corresponds to `x0000000`

### Which version(s) does this affect? (Environment, OS, etc...)
6.0.0-beta.1-

### PR DESCRIPTION
**Pull request by mosmartin**
### What was the problem?

This PR resolves #8471

### How was it solved?

Updated the `_isMainchainOrNative (...)` function to only support the lsk token.

### How was it tested?

Updated unit tests

### CODE REVIEW

File: framework/test/unit/modules/token/stores/supported_tokens.spec.ts [hunk 1]
```diff
@@ -58,6 +58,8 @@ describe('SupportedTokensStore', () => {
 		});
 
 		it('should return true if tokenID is LSK', async () => {
+			await store.set(context, Buffer.from([1, 0, 0, 0]), { supportedTokenIDs: [] });
```
Comment (shuse2) [inline framework/test/unit/modules/token/stores/supported_tokens.spec.ts:61]: we should add a test where it's different token from mainchain. in this example, something like `[1,0,0,0,0,0,0,1]`

File: framework/src/modules/token/stores/supported_tokens.ts [hunk 2]
```diff
@@ -197,6 +197,9 @@ export class SupportedTokensStore extends BaseStore<SupportedTokensStoreData> {
 			return true;
 		}
 
-		return chainID[0] === this._ownChainID[0] && getMainchainID(chainID).equals(chainID);
+		return (
+			chainID[0] === this._ownChainID[0] &&
+			getMainchainID(chainID).equals(Buffer.alloc(CHAIN_ID_LENGTH, 0))
```
Comment (shuse2) [inline framework/src/modules/token/stores/supported_tokens.ts:202]: Checking if the tokenID is LSK or not should be
* network ID is the same `chainID[0] === this._ownChainID[0]`
* chain ID is the mainchain ID `getMainchainID(chainID).equals(chainID)`
* tokenID is `[0,0,0,0]` which is mising here

**Review by shuse2 — CHANGES_REQUESTED** [review]

File: framework/src/modules/token/stores/supported_tokens.ts (hunk 2 above)
Comment (mosmartin) [reply framework/src/modules/token/stores/supported_tokens.ts:202]: @shuse2 is the value you mention of  tokenID `[0,0,0,0]` supposed to be the `chain id` or `local id` as token id has 8 bytes `[0, 0, 0, 0, 0, 0, 0, 0]`?

Comment (shuse2) [reply framework/src/modules/token/stores/supported_tokens.ts:202]: what i meant is local id. so it should be `[N, 0, 0, 0, 0, 0, 0, 0]`

**Review by bobanm — APPROVED** [review]
I propose to change the logic for detecting LSK token into:

```typescript
const networkId = this._ownChainID.slice(0, 1);
const lskTokenId = Buffer.concat([networkId, Buffer.alloc(CHAIN_ID_LENGTH + LOCAL_ID_LENGTH - 1, 0)]);

return tokenID.equals(lskTokenId);
```

And now that the function returns false for any mainchain token other than LSK, it would be good to rename method `_isMainchainOrNative()` into `_isLskOrNativeToken()`

But I'm also fine if it remains as it is 🙂

File: framework/test/unit/modules/token/stores/supported_tokens.spec.ts [hunk 3]
```diff
@@ -58,11 +58,19 @@ describe('SupportedTokensStore', () => {
...
+
 			await expect(
 				store.isSupported(context, Buffer.from([1, 0, 0, 0, 0, 0, 0, 0])),
 			).resolves.toBeTrue();
 		});
 
+		it('should return false if token is different from mainchain', async () => {
```
Comment (shuse2) [inline framework/test/unit/modules/token/stores/supported_tokens.spec.ts:68]:
```suggestion
		it('should return false if token is from mainchain but not LSK', async () => {
```

File: framework/src/modules/token/stores/supported_tokens.ts [hunk 4]
```diff
@@ -197,6 +197,11 @@ export class SupportedTokensStore extends BaseStore<SupportedTokensStoreData> {
...
 		}
 
-		return chainID[0] === this._ownChainID[0] && getMainchainID(chainID).equals(chainID);
+		return (
+			chainID[0] === this._ownChainID[0] &&
+			getMainchainID(chainID).equals(chainID) &&
+			tokenID ===
+				Buffer.concat([Buffer.from([chainID[0], 0, 0, 0]), Buffer.alloc(LOCAL_ID_LENGTH, 0)])
+		);
 	}
```
Comment (shuse2) [inline framework/src/modules/token/stores/supported_tokens.ts:206]: i missed on the last review, but Buffer must use `equals` to compare

**Review by shuse2 — CHANGES_REQUESTED** [review]

**Review by shuse2 — APPROVED** [review]

**Pull request merged**